Common/src/toolboxes/C1DInterpolation.cpp

```diff
@@ -0,0 +1,202 @@
+/*!
+ * \file inlet_interpolation_functions.cpp
+ * \brief Inlet_interpolation_functions
+ * \author Aman Baig
+ * \version 7.0.1 "Blackbird"
+ *
+ * SU2 Project Website: https://su2code.github.io
+ *
+ * The SU2 Project is maintained by the SU2 Foundation 
+ * (http://su2foundation.org)
+ *
+ * Copyright 2012-2019, SU2 Contributors (cf. AUTHORS.md)
+ *
+ * SU2 is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * SU2 is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with SU2. If not, see <http://www.gnu.org/licenses/>.
+ */
+ 
+
+#include "../../include/toolboxes/C1DInterpolation.hpp"
+
+/*--- Function for evaluating the value for akima spline ---*/
+su2double CAkimaInterpolation::EvaluateSpline(su2double Point_Interp){
+    Point_Match = true;
+    int i =0, j=this->n-1;
+
+    while (j-i>1){ 
+        int m=(i+j) / 2 ; 
+            if (Point_Interp>this->x[m]) {i=m;}
+            else {j=m;} 
+    }
+
+    su2double h=Point_Interp-this->x[i] ;
+    return this->y[i]+h*(this->b[i]+h*(this->c[i]+h*this->d[i])) ;
```

Comment:
You don't need the `this` pointer, methods of CAkimaInterpolation have access to member variables.
It creates a bit of confusion when used next to `>` or `-`.

Reply:
Yes I did that because you pointed out not to use the 'side effect' of updating the variables like this.